src/general/ModLoader.cs

```diff
@@ -0,0 +1,320 @@
+        {
+            return;
+        }
+
+        var selectedItem = unloadedItemList.GetSelectedItems()[0];
+        loadedModList.Add(modList[selectedItem]);
+        loadedItemList.AddItem(modList[selectedItem].ModName);
+        modList.RemoveAt(selectedItem);
+        unloadedItemList.RemoveItem(selectedItem);
+    }
+
+    private void OnMoveToUnloadPressed()
+    {
+        GUICommon.Instance.PlayButtonPressSound();
+        if (loadedItemList.GetSelectedItems().Length <= 0)
+        {
+            return;
+        }
+
+        var selectedItem = loadedItemList.GetSelectedItems()[0];
+        modList.Add(loadedModList[selectedItem]);
+        unloadedItemList.AddItem(loadedModList[selectedItem].ModName);
+        loadedModList.RemoveAt(selectedItem);
+        loadedItemList.RemoveItem(selectedItem);
+    }
+
+    private void OnRefreshPressed()
+    {
+        GUICommon.Instance.PlayButtonPressSound();
+        modList.Clear();
+        loadedModList.Clear();
+        unloadedItemList.Clear();
+        loadedItemList.Clear();
+        DirectoryInfo modFolder = Directory.CreateDirectory($"{Directory.GetCurrentDirectory()}\\mods");
+        foreach (DirectoryInfo currentMod in modFolder.EnumerateDirectories())
+        {
+            if (!File.Exists(currentMod.FullName + "/mod_info.json"))
+            {
+                continue;
+            }
+
+            var currentModInfo =
+                JsonConvert.DeserializeObject<ModInfo>(ReadJSONFile(currentMod.FullName + "/mod_info.json"));
+
+            if (currentModInfo.AutoLoad)
+            {
+                continue;
+            }
+
+            currentModInfo.Location = currentMod.FullName;
+            modList.Add(currentModInfo);
+            unloadedItemList.AddItem(currentModInfo.ModName);
+        }
+    }
+
+    private void OnLoadPressed()
+    {
+        GUICommon.Instance.PlayButtonPressSound();
+        if (loadedModList.Count <= 0)
+        {
+            return;
+        }
+
+        foreach (ModInfo currentMod in loadedModList)
+        {
+            if (string.IsNullOrEmpty(currentMod.Dll))
+            {
+                if (File.Exists(currentMod.Location + "/" + currentMod.Dll))
+                {
+                    Assembly.LoadFile(currentMod.Location + "/" + currentMod.Dll);
+                }
+            }
+
+            if (!File.Exists(currentMod.Location + "/mod.pck"))
+            {
+                GD.Print("Fail to find mod file: " + currentMod.ModName);
+                continue;
+            }
+
+            if (ProjectSettings.LoadResourcePack(currentMod.Location + "/mod.pck", true))
+            {
+                GD.Print("Loaded mod: " + currentMod.ModName);
+            }
+            else
+            {
+                GD.Print("Failed to load mod: " + currentMod.ModName);
+            }
+        }
+
+        GD.Print("All mods loaded");
+        SceneManager.Instance.ReturnToMenu();
```

Comment:
I'm a bit suspicious why this is done and not a normal return to the menu? Does it has to do with mods otherwise not being able to overwrite things properly?

Reply:
What is the normal way to return to the main menu?

Reply:
Trigger the return callback like with the back button there is in the mod manager.
Also the same thing in the options menu and save load menu.

Reply:
Yeah i do that as if i don't mods don't load properly